meteor/client/ui/ClockView/OverlayScreen.tsx

```diff
@@ -0,0 +1,209 @@
+							this.subscribe(PubSub.rundowns, {
+								playlistId: playlist._id,
+							})
+
+							this.autorun(() => {
+								const rundownIds = playlist!.getRundownIDs()
+
+								this.subscribe(PubSub.segments, {
+									rundownId: { $in: rundownIds },
+								})
+								this.subscribe(PubSub.parts, {
+									rundownId: { $in: rundownIds },
+								})
+								this.subscribe(PubSub.partInstances, {
+									rundownId: { $in: rundownIds },
+									reset: { $ne: true },
+								})
+
+								this.autorun(() => {
+									let playlist = RundownPlaylists.findOne(this.props.playlistId, {
+										fields: {
+											_id: 1,
+											currentPartInstanceId: 1,
+											nextPartInstanceId: 1,
+											previousPartInstanceId: 1,
+										},
+									}) as
+										| Pick<
+												RundownPlaylist,
+												| '_id'
+												| 'currentPartInstanceId'
+												| 'nextPartInstanceId'
+												| 'previousPartInstanceId'
+												| 'getSelectedPartInstances'
+										  >
+										| undefined
+									const { nextPartInstance, currentPartInstance } = playlist!.getSelectedPartInstances()
+									this.subscribe(PubSub.pieceInstances, {
+										partInstanceId: {
+											$in: [currentPartInstance?._id, nextPartInstance?._id],
+										},
+									})
+								})
+							})
+						}
+					})
+				}
+
+				componentWillUnmount() {
+					super.componentWillUnmount()
+					document.body.classList.remove('transparent')
+				}
+
+				render() {
+					const { playlist, segments, showStyleBaseId, t } = this.props
+
+					if (playlist && this.props.playlistId && this.props.segments && showStyleBaseId) {
+						let currentPart: PartUi | undefined
+						let currentSegment: SegmentUi | undefined
+						for (const segment of segments) {
+							if (segment.items) {
+								for (const item of segment.items) {
+									if (item.instance._id === playlist.currentPartInstanceId) {
+										currentSegment = segment
+										currentPart = item
+									}
+								}
+							}
+						}
+						let currentPartDuration: number | null = null
+						if (currentPart) {
+							currentPartDuration = currentPart.renderedDuration || currentPart.instance.part.expectedDuration || 0
+							currentPartDuration += -1 * (currentPart.instance.timings?.duration || 0)
+							if (!currentPart.instance.timings?.duration && currentPart.instance.timings?.startedPlayback) {
+								currentPartDuration += -1 * (getCurrentTime() - currentPart.instance.timings.startedPlayback)
+							}
+						}
+
+						let nextPart: PartUi | undefined
+						let nextSegment: SegmentUi | undefined
+						for (const segment of segments) {
+							if (segment.items) {
+								for (const item of segment.items) {
+									if (item.instance._id === playlist.nextPartInstanceId) {
+										nextSegment = segment
+										nextPart = item
+									}
+								}
+							}
+						}
+
+						// const overUnderClock = playlist.expectedDuration
```

Comment:
Dead code. Should either be deleted, or commented to tell why it should be saved for later

Reply:
I've added some comments to tell why I left these parts in for future reference.